lib/idn.c

```diff
@@ -50,6 +50,63 @@
 #include "curl_memory.h"
 #include "memdebug.h"
 
+/* for macOS and iOS targets */
+#if defined(HAVE_APPLE_IDN)
+#include <unicode/uidna.h>
+
+static CURLcode mac_idn_to_ascii(const char *in, char **out)
+{
+  UErrorCode err = U_ZERO_ERROR;
+  UIDNA* idna = uidna_openUTS46(UIDNA_CHECK_BIDI, &err);
+  if(U_FAILURE(err)) {
+    return CURLE_OUT_OF_MEMORY;
+  }
+  else {
+    UIDNAInfo info = UIDNA_INFO_INITIALIZER;
+    char buffer[256] = {0};
+    int32_t len = uidna_nameToASCII_UTF8(idna, in, -1, buffer,
```

Comment:
This is showing the warning (also for the next similar line):
```
./lib/idn.c:67:13: warning: unused variable 'len' [-Wunused-variable]
    int32_t len = uidna_nameToASCII_UTF8(idna, in, -1, buffer,
            ^
./lib/idn.c:93:13: warning: unused variable 'len' [-Wunused-variable]
    int32_t len = uidna_nameToUnicodeUTF8(idna, in, -1, buffer,
            ^
2 warnings generated.
```

Is `len` supposed to be used later? Or should these be cast to `(void)` to ignore the return value?

Reply:
Well, we don't use the len currently:

`*out = strdup(buffer);
`

We could use a strdup variant which takes a length instead of relaying on the NULL byte.
Or just cast to void. What would you prefer?

Reply:
These APIs return the length to let us allocate a buffer dynamically. Since we use fixed ones (also for WinIDN and libidn2), we can just `(void)` them, I think.

Reply:
Okay. I'll change it.